src/main/java/ch/njol/skript/effects/EffPlaySound.java

```diff
@@ -47,186 +46,139 @@
-			Skript.registerEffect(EffPlaySound.class,
-					"play sound[s] %strings% [(in|from) %-soundcategory%] " +
-							"[(at|with) volume %-number%] [(and|at|with) pitch %-number%] at %locations% [(to|for) %-players%]",
-					"play sound[s] %strings% [(in|from) %-soundcategory%] " +
-							"[(at|with) volume %-number%] [(and|at|with) pitch %-number%] [(to|for) %players%] [(at|from) %-locations%]");
-		} else {
-			Skript.registerEffect(EffPlaySound.class,
-					"play sound[s] %strings% [(at|with) volume %-number%] " +
-							"[(and|at|with) pitch %-number%] at %locations% [(to|for) %-players%]",
-					"play sound[s] %strings% [(at|with) volume %-number%] " +
-							"[(and|at|with) pitch %-number%] [(to|for) %players%] [(at|from) %-locations%]");
-		}
+		Skript.registerEffect(EffPlaySound.class,
+				"play sound[s] %strings% [(in|from) %-soundcategory%] " +
+						"[(at|with) volume %-number%] [(and|at|with) pitch %-number%] at %locations% [(to|for) %-players%]",
+				"play sound[s] %strings% [(in|from) %-soundcategory%] " +
+						"[(at|with) volume %-number%] [(and|at|with) pitch %-number%] [(to|for) %players%] [(at|from) %-locations%]"
+		);
 	}
 
-	@SuppressWarnings("null")
+	@SuppressWarnings("NotNullFieldNotInitialized")
 	private Expression<String> sounds;
 	@Nullable
 	private Expression<SoundCategory> category;
 	@Nullable
-	private Expression<Number> volume, pitch;
+	private Expression<Number> volume;
+	@Nullable
+	private Expression<Number> pitch;
 	@Nullable
 	private Expression<Location> locations;
 	@Nullable
 	private Expression<Player> players;
 
 	@Override
-	@SuppressWarnings({"unchecked", "null"})
+	@SuppressWarnings("unchecked")
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		sounds = (Expression<String>) exprs[0];
-		if (SOUND_CATEGORIES_EXIST) {
-			category = (Expression<SoundCategory>) exprs[1];
-			volume = (Expression<Number>) exprs[2];
-			pitch = (Expression<Number>) exprs[3];
-			if (matchedPattern == 0) {
-				locations = (Expression<Location>) exprs[4];
-				players = (Expression<Player>) exprs[5];
-			} else {
-				players = (Expression<Player>) exprs[4];
-				locations = (Expression<Location>) exprs[5];
-			}
+		category = (Expression<SoundCategory>) exprs[1];
+		volume = (Expression<Number>) exprs[2];
+		pitch = (Expression<Number>) exprs[3];
+		if (matchedPattern == 0) {
+			locations = (Expression<Location>) exprs[4];
+			players = (Expression<Player>) exprs[5];
 		} else {
-			volume = (Expression<Number>) exprs[1];
-			pitch = (Expression<Number>) exprs[2];
-			if (matchedPattern == 0) {
-				locations = (Expression<Location>) exprs[3];
-				players = (Expression<Player>) exprs[4];
-			} else {
-				players = (Expression<Player>) exprs[3];
-				locations = (Expression<Location>) exprs[4];
-			}
+			players = (Expression<Player>) exprs[4];
+			locations = (Expression<Location>) exprs[5];
 		}
+		
 		return true;
```

Comment:
Should we block init if players and locations are null?

Reply:
I'll change that